src/OpenTelemetry/Trace/TracerProviderBuilderExtensions.cs

```diff
@@ -54,6 +54,22 @@ public static TracerProviderBuilder SetResource(this TracerProviderBuilder trace
             return tracerProviderBuilder;
         }
 
+        /// <summary>
+        /// Registers a callback that will be executed to configure the <see cref="Resource"/> for the <see cref="TracerProvider"/>.
+        /// </summary>
+        /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/>.</param>
+        /// <param name="configureResource"><see cref="ResourceBuilder"/>.</param>
+        /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
+        public static TracerProviderBuilder ConfigureResource(this TracerProviderBuilder tracerProviderBuilder, Action<ResourceBuilder> configureResource)
```

Comment:
Lets hold on to exposing this API. 
1. The behavior is a bit diff. with this. In you Usecase1 example, the default TelemetrySDKResource will be automatically part of the final resource. In Usecase2 example, TelemetrySdkResource is not part of final resource.
2. SetResource() API was clear that *only* one resource can be set. Calling SetResource() again will override the first one. ConfigureResource() - its a bit unclear...the current implementation is last-one-wins. But users could be misunderstanding that multiple calls to ConfigureResource() would be chained. (its typical in .net core, right?)

Lets merge this PR without ConfigureResource, and add ConfigureResource option later after more discussion. The ResourceBuilder approach solves a good amount of current problems, so ConfigureResource API is not a necessity now.

Reply:
Cool I'll remove it for now pending more discussion.

Just a little more info on the current design...

I envisioned it working sort of like [`WebHost.CreateDefaultBuilder`](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-5.0#set-up-a-host) where it gives you all the default stuff automatically (TelemetrySdk + anything else the spec says should be in there) and then allows you to (very easily) add extra things in. That's this pattern:

```csharp
            var builder = Sdk.CreateTracerProviderBuilder()
                 .ConfigureResource(r => r
                    .AddService("MyService")
                    .AddAttributes(new Dictionary<string, object>
                    {
                        ["service.datacenter"] = 1,
                    }));
```

But if you want to completely override the defaults, there is a `Clear` like this:

```csharp
            var builder = Sdk.CreateTracerProviderBuilder()
                 .ConfigureResource(r => r
                    .Clear()
                    .AddAttributes(new Dictionary<string, object>
                    {
                        ["resource.only.item"] = 1,
                    }));
```

Reply:
yea i am not opposed to adding back ConfigureResource subsequently- just want to keep a single option initially and bring it back as we get more time to fine tune it.

Reply:
Removed (and updated API in the description)